### PR TITLE
Fixed/improved build

### PR DESCRIPTION
**Pull request by palfrey**
* Used Homebrew cask-versions to install zulu7 (because no Java 7 in homebrew anymore since https://github.com/Homebrew/homebrew-cask-versions/pull/3914)
* Installed [maven-wrapper](https://github.com/takari/maven-wrapper) so I could lock down the Maven version to a 1.6 compatible one
* ~~Switched to HTTP repositories as the HTTPS ones need TLS 1.2 which JDK 1.6 doesn't have~~
* Fixed some old bits of Travis workarounds and cached everything
* Split out deploy builds (only on master) from test builds (everything)

### CODE REVIEW

**Comment by palfrey** [comment]
I'm still seeing "Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty" on the deploys, but can't tell if that's my builds or not because of the secure parameters.

File: pom.xml [hunk 1]
```diff
@@ -50,14 +50,29 @@
...
+      <url>http://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
     </repository>
     <snapshotRepository>
       <id>sonatype-ossrh</id>
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+      <url>http://oss.sonatype.org/content/repositories/snapshots</url>
     </snapshotRepository>
   </distributionManagement>
 
+  <!-- needed for JDK 1.6 which doesn't support TLS 1.2 -->
```
Comment (gjoseph) [inline pom.xml:61]: Ouch! Really? Maybe it's time to forget about JDK 1.6 for this project anyway. (if we want, we can still have `target` set to 1.6, but use 1.7+ to build?)

Comment (palfrey) [reply pom.xml]: I've been able to setup JDK 1.7 instead, and get rid of these POM changes

Comment (palfrey) [reply pom.xml]: I've been able to setup JDK 1.7 instead, and get rid of these POM changes

**Comment by gjoseph** [comment]
Hi @palfrey sorry for the slow response -- as you may have gathered, I haven't had capacity to maintain this project in a long time. I'm finally officially retiring it. See README for possible alternatives, if you're still needing this. ✌️

**Comment by palfrey** [comment]
Thanks for all your work on this.

**Pull request closed without merging**